jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/BytecodeAnnotator.java

```diff
@@ -103,14 +172,7 @@ private static void annotateBytecode(
         for (Integer param : params) {
           int paramNum = isStatic ? param : param - 1;
           // Add a @Nonnull annotation on this parameter.
-          if (method.visibleParameterAnnotations == null
-              || method.visibleParameterAnnotations.length < paramNum
-              || method.visibleParameterAnnotations[paramNum] == null) {
-            method.visitParameterAnnotation(paramNum, javaxNonnullDesc, true);
-          } else {
-            addAnnotationIfNotPresent(
-                method.visibleParameterAnnotations[paramNum], javaxNonnullDesc);
-          }
+          method.visitParameterAnnotation(paramNum, nonnullDesc, visible);
```

Comment:
Does the same visibility rules hold for both nullable and non-null annotations?

Reply:
In the two cases we support? Yes. We could make this more flexible in the future, I guess, but seems like a needless early optimization.

Reply:
LGTM